src/Aspire.Hosting.Certbot/CertbotBuilderExtensions.cs

```diff
@@ -57,27 +58,74 @@ public static IResourceBuilder<CertbotResource> AddCertbot(
-            .WithHttpEndpoint(port: 80, targetPort: 80, name: CertbotResource.HttpEndpointName)
+            .WithVolume(CertbotResource.CertificatesVolumeName, CertbotResource.CertificatesPath);
+    }
+
+    /// <summary>
+    /// Configures Certbot to use the HTTP-01 challenge for domain validation.
+    /// </summary>
+    /// <param name="builder">The Certbot resource builder.</param>
+    /// <param name="port">The host port to publish for the HTTP-01 challenge. Defaults to 80.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// <para>
+    /// The HTTP-01 challenge requires Certbot to be accessible on port 80 from the internet.
+    /// This method configures the container to listen on the specified port and adds the necessary
+    /// command-line arguments for standalone mode.
+    /// </para>
+    /// <example>
+    /// <code lang="csharp">
+    /// var certbot = builder.AddCertbot("certbot", domain, email)
+    ///     .WithHttp01Challenge(port: 8080);
+    /// </code>
+    /// </example>
+    /// </remarks>
+    public static IResourceBuilder<CertbotResource> WithHttp01Challenge(
+        this IResourceBuilder<CertbotResource> builder,
+        int? port = 80)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        return builder
+            .WithHttpEndpoint(port: port, targetPort: 80, name: CertbotResource.HttpEndpointName)
             .WithExternalHttpEndpoints()
             .WithArgs(context =>
             {
+                var resource = (CertbotResource)context.Resource;
                 context.Args.Add("certonly");
                 context.Args.Add("--standalone");
                 context.Args.Add("--non-interactive");
                 context.Args.Add("--agree-tos");
                 context.Args.Add("-v");
                 context.Args.Add("--keep-until-expiring");
-                // Fix permissions so non-root containers can read the certs
-                context.Args.Add("--deploy-hook");
-                context.Args.Add("chmod -R 755 /etc/letsencrypt/live && chmod -R 755 /etc/letsencrypt/archive");
                 context.Args.Add("--email");
                 context.Args.Add(resource.EmailParameter);
                 context.Args.Add("-d");
                 context.Args.Add(resource.DomainParameter);
             });
     }
 
+    /// <summary>
+    /// Configures Certbot to fix permissions on certificate files so non-root containers can read them.
+    /// </summary>
+    /// <param name="builder">The Certbot resource builder.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// This method adds a deploy hook that changes permissions on the certificate directories
+    /// to allow non-root containers to read the certificates.
+    /// </remarks>
+    public static IResourceBuilder<CertbotResource> WithPermissionFix(
```

Comment:
This name makes it sound like a bug. This is allowing non admin read access. We should just always to this.